src/main/java/com/walmartlabs/x12/X12TransactionSet.java

```diff
@@ -24,62 +29,91 @@
+     * and 810 (invoice).
      * 
-     * @return the ST01 segment value 
+     * @return the ST01 segment value
      */
     String getTransactionSetIdentifierCode();
-    
+
     void setTransactionSetIdentifierCode(String transactionSetIdentifierCode);
 
     /**
-     * The ST02 segment element contains the control number. This should match
-     * the control number on the corresponding transaction trailer segment.
+     * The ST02 segment element contains the control number. This should match the
+     * control number on the corresponding transaction trailer segment.
      * 
-     * @return the ST02 segment value 
+     * @return the ST02 segment value
      */
-    String getHeaderControlNumber() ;
-    
+    String getHeaderControlNumber();
+
     void setHeaderControlNumber(String headerControlNumber);
 
     /**
-     * The SE01 segment element contains the number of segments that 
-     * are in this transaction.
+     * The SE01 segment element contains the number of segments that are in this
+     * transaction.
      * 
-     * @return the SE01 segment value 
+     * @return the SE01 segment value
      */
     Integer getExpectedNumberOfSegments();
-    
+
     void setExpectedNumberOfSegments(Integer expectedNumberOfSegments);
 
     /**
-     * The SE02 segment element contains the control number. This should match
-     * the control number on the corresponding transaction header segment.
+     * The SE02 segment element contains the control number. This should match the
+     * control number on the corresponding transaction header segment.
      * 
-     * @return the SE02 segment value 
+     * @return the SE02 segment value
      */
     String getTrailerControlNumber();
-    
+
     void setTrailerControlNumber(String trailerControlNumber);
-    
+
     /**
      * The CTT01 segment element contains the transaction count. 
-     * This is an optional segment lien
+     * This is an optional segment line
      * 
-     * @return the CTT01 segment value 
+     * @return the CTT01 segment value
      */
     Integer getTransactionLineItems();
 
     void setTransactionLineItems(Integer transactionLineItems);
+
+    /**
+     * any looping errors will be stored in this list 
+     * 
+     * possible errors include
+     * orphan loops and those without proper indexing
+     * 
+     * @return list of {@link X12ErrorDetail}
+     */
+    default List<X12ErrorDetail> getLoopingErrors() {
+        return Collections.emptyList();
+    }
+
+    /**
+     * check to determine if loop parsing was valid
+     * @return true if there are looping errors otherwise false
+     */
+    default boolean isLoopingValid() {
+        return false;
+    }
+
+    /**
+     * check to see if the transaction set supports looping
+     * 
+     * @return true if this type of transaction set supports loops otherwise false
+     */
+    default boolean hasLooping() {
```

Comment:
🤔 
considering the name `supportsLooping` instead of `hasLooping`
the former (upon further review) seems to better indicate that this helps a user determine if the transaction set supports HL instead of whether there are loops available

```
if (txSet.supportsLooping()) {
     // need to get access to loops
}
```